Source/UnitTests/ExecutionEngineTests/ProofIsolation.cs

```diff
@@ -11,6 +11,86 @@ namespace ExecutionEngineTests
   [TestFixture]
   public class ProofIsolation
   {
+    [Test()]
+    public void TestNameDiscarding()
+    {
+      var procedure1 = @"
+type Wicket;
+const w : Wicket;
+function age(Wicket) returns (int);
+axiom age(w) == 7;
+
+var favorite: Wicket;
+var alternative: Wicket;
+
+type Barrel a;
+type RGBColor;
+const unique red: RGBColor;
+const unique green: RGBColor;
+const unique blue: RGBColor;
+
+const m: [Barrel Wicket] Wicket;
+const n: <a> [Barrel a] a;
+
+type MySynonym a = int;
+type ComplicatedInt = MySynonym (MySynonym bool);
+
+procedure M(x: int, coloredBarrel: Barrel RGBColor)
+  requires x == 2; 
+  modifies favorite;
+  ensures age(favorite) == 42; 
+{
+  var y: ComplicatedInt;
+  favorite := alternative;
+  assert y == 3;
+}
+";
+      
+      var procedure2 = @"
+type Wicket2;
+const w2 : Wicket2;
+function age2(Wicket2) returns (int);
+axiom age2(w2) == 7;
+
+var favorite2: Wicket2;
+var alternative2: Wicket2;
+
+type Barrel2 a;
+type RGBColor2;
+const unique red2: RGBColor2;
+const unique green2: RGBColor2;
+const unique blue2: RGBColor2;
+
+const m2: [Barrel2 Wicket2] Wicket2;
+const n2: <a> [Barrel2 a] a;
```

Comment:
Can you also rename the type parameter `a`, the procedure `x`, the input parameter `x` and the variable `y` ?

Reply:
Good call :)